### PR TITLE
hardening/1016_add_cartodb_doc

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #1016 
* No tests are required
* Assignee @pcoello25 , but LGTM from @gtorodelvalle is also required

### CODE REVIEW

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+
+###<a name="section1.2"></a>Mapping Flume events to CartoDB data structures
+CartoDB is based on [PostgreSQL](http://www.postgresql.org/) and [PostGIS](http://postgis.net/) extensions. It organizes the data in databases (one per organization), schemas (una per project within an organization) and tables (a schema may have one or more tables). Such organization is exploited by `NGSICartoDBSink` each time a Flume event is going to be persisted.
+
+[Top](#top)
+
+####<a name="section1.2.1"></a>PostgreSQL databases and schemas naming conventions
+PostgreSQL databases and schemas are already created by CartoDB upon organization and project request, respectively. Thus, it is up to CartoDB to define the naming conventions for these elements.
+
+Here it is asseumed the notified/default FIWARE service maps the PostgreSQL schema/project name.
```
Comment (pcoello25) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: asseumed -> assumed

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+
+[Top](#top)
+
+####<a name="section1.2.2"></a>PostgreSQL tables naming conventions
+The name of these tables depends on the configured data model (see the [Configuration](#section2.1) section for more details):
+
+* Data model by service path (`data_model=dm-by-service-path`). As the data model name denotes, the notified FIWARE service path (or the configured one as default in [`NGSIRestHandler`](.ngsi_rest_handler.md)) is used as the name of the table. This allows the data about all the NGSI entities belonging to the same service path is stored in this unique table. The only constraint regarding this data model is the FIWARE service path cannot be the root one (`/`).
+* Data model by entity (`data_model=dm-by-entity`). For each entity, the notified/default FIWARE service path is concatenated to the notified entity ID and entityType in order to compose the table name. The concatenation charated is `_` (underscore). If the FIWARE service path is the root one (`/`) then only the entity ID and type are concatenated.
```
Comment (pcoello25) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: charated -> character

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+* Data model by attribute (`data_model=dm-by-attribute`). For each entity and entity's attribute, the notified/default FIWARE service path, the notified entityId and entityType, and the attribute name and type are concatenated in order to compose the table name. The concatenation charated is `_` (underscore). If the FIWARE service path is the root one (`/`) then only the entity ID and type and the attribute name and type are concatenated.
```
Comment (pcoello25) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: charated again :yum:

**Comment by pcoello25** [comment]
LGTM, passing the ball to @gtorodelvalle :)

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+###<a name="section1.1"></a>Mapping NGSI events to flume events
+Notified NGSI events (containing context data) are transformed into Flume events (such an event is a mix of certain headers and a byte-based body), independently of the NGSI data generator or the final backend where it is persisted.
+
+This is done at the cygnus-ngsi Http listeners (in Flume jergon, sources) thanks to [`NGSIRestHandler`](./orion_rest_handler.md). Once translated, the data (now, as a Flume event) is put into the internal channels for future consumption (see next section).
+
+[Top](#top)
+
+###<a name="section1.2"></a>Mapping Flume events to CartoDB data structures
+CartoDB is based on [PostgreSQL](http://www.postgresql.org/) and [PostGIS](http://postgis.net/) extensions. It organizes the data in databases (one per organization), schemas (una per project within an organization) and tables (a schema may have one or more tables). Such organization is exploited by `NGSICartoDBSink` each time a Flume event is going to be persisted.
+
```
Comment (gtorodelvalle) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: `schemas (una per project within an organization)`, 2 issues here:

1. `una` -> `one`
2. Is it really one per "project" or one per "user"?

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md [hunk 5]
```diff
@@ -0,0 +1,455 @@
...
+[Top](#top)
+
+####<a name="section1.2.1"></a>PostgreSQL databases and schemas naming conventions
+PostgreSQL databases and schemas are already created by CartoDB upon organization and project request, respectively. Thus, it is up to CartoDB to define the naming conventions for these elements.
```
Comment (gtorodelvalle) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: What is a `project` in the context of CartoDB?

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md [hunk 6]
```diff
@@ -0,0 +1,455 @@
...
+A single insert is composed for each notified entity, containing such insert the following fields:
+
+* `recvTime`: UTC timestamp in human-redable format ([ISO 8601](http://en.wikipedia.org/wiki/ISO_8601)).
+* `fiwareServicePath`: Notified fiware-servicePath, or the default configured one if not notified.
+* `entityId`: Notified entity identifier.
+* `entityType`: Notified entity type.
+* `the_geom`: The current geolocation point, i.e. containing longitude and latitude. It must be of type PostGIS Geometry, it can be created for instance using [`ST_Point()`](http://postgis.net/docs/ST_Point.html). The data used for composing the PostGIS Geometry is obtained from a special notified attribute:
+    * Either of type `geo:point`.
+    * Either having associated a `location` metadata of type `string` and value `WGS84`.
+* For each not-geolocated attribute, the insert will contain two additional fields, one for the value, named `<attrName>`, and another for the metadata, named `<attrName>_md`).
```
Comment (gtorodelvalle) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: Just out of my curiosity, is the same information stored no mater the data model? Or depending on the data model the information included in the table names is not included?

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md (hunk 4 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: It depends on what we consider a "project" and a "user". Let's discuss it offline ;)

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md (hunk 6 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: The data stored is always the same. So, data model, and thus the table name, is just a way of concentrating more or less NGSI data in more or less tables.

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md (hunk 5 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1022/commits/273a671ffa3dd8e412fccbf6abffd91742d05cdd as discussed offline.

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md (hunk 4 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1022/commits/273a671ffa3dd8e412fccbf6abffd91742d05cdd as discussed offline.

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md (hunk 3 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1022/commits/273a671ffa3dd8e412fccbf6abffd91742d05cdd

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md (hunk 2 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1022/commits/273a671ffa3dd8e412fccbf6abffd91742d05cdd

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md (hunk 1 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1022/commits/273a671ffa3dd8e412fccbf6abffd91742d05cdd

**Comment by gtorodelvalle** [comment]
LGTM and merging ;)

**Pull request merged**